### PR TITLE
Test coverage for Keep the URIs in the metrics tag with 302 and 404 patterns

### PR DESCRIPTION
**Pull request by jcarranzan**
### Summary

Test coverage for QUARKUS-4430  https://github.com/quarkusio/quarkus/pull/39583

TP: https://github.com/quarkus-qe/quarkus-test-plans/pull/177 
Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: monitoring/micrometer-prometheus/src/main/java/io/quarkus/ts/micrometer/prometheus/PathPatternResource.java
Comment (michalvavrik) [inline monitoring/micrometer-prometheus/src/main/java/io/quarkus/ts/micrometer/prometheus/PathPatternResource.java:30]:
```suggestion
        return Response
                .status(Response.Status.MOVED_PERMANENTLY)
                .entity("Moved resource with ID: " + id)
                .build();
```
CI says you expect 301 but respond with 200

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (michalvavrik) [inline monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:78]: `.redirects().follow(false)`

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (michalvavrik) [inline monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:55]:
```suggestion
       thenMetricIsExposedInServiceEndpoint(301, "REDIRECTION", MOVED_ENDPOINT);
```

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (michalvavrik) [inline monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:36]:
```suggestion
        thenMetricIsExposedInServiceEndpoint(404, "CLIENT_ERROR", "NOT_FOUND");
```
because actual response is `http_server_requests_seconds_count{method="GET",outcome="CLIENT_ERROR",status="404",uri="NOT_FOUND"} 1.0` and `/test/not-found` has no path param. I suggest to complement this test with `testNotFoundEndopintDynamic` (using your terminology, not sure it's alright TBH!) so that you can test `NOT_FOUND_ENDPOINT` in URI.

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (michalvavrik) [inline monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:48]: This fails because you call it with `/test/` and they record `/test` and so, metrics are there for `/test` and not for `/test/`. That alone could be acceptable, however I mentioned that it is also treated as 'not constant" and URI is `/test/` because they to plain equals. I commented on this here https://github.com/quarkusio/quarkus/pull/39583/files#r1636726541, however I don't know what to do. Maybe you should open an issue. Or not.

Anyway this completely undocumented feature is strange, I think you need to act somehow, like request documentation so that expectations are clear.

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (michalvavrik) [inline monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:42]:
```suggestion
        thenMetricIsExposedInServiceEndpoint(302, "REDIRECTION", "REDIRECTION");
```
fails because there is no template; please create additional dynamic redirect test that will test expected URI to be `REDIRECT_ENDPOINT`.

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (jcarranzan) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:60]: oh it's true, I just remember that I used the same `redirects.follow(false)` for webauthn tests. Thank you for remind me that @michalvavrik

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (jcarranzan) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:48]: Yes, I agree with opening an issue and there request documentation with expectation behavior.

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (jcarranzan) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:36]: Currently with the Quarkus code, for 404, as you mentioned the response is `http_server_requests_seconds_count{method="GET",outcome="CLIENT_ERROR",status="404",uri="NOT_FOUND"} 1.0 `  and I've tried diferent ways but not sure if it's possible to do coincide them:  `pathInfo` with `pathInfo` for returning the uri path , according source code:
` } else if (code == 404) {
                if (isTemplatedPath(pathInfo, initialPath)) {
                    return Tag.of("uri", pathInfo);
                } else {
                    return URI_NOT_FOUND;
                }
            }.`

Comment (michalvavrik) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:36]: I don't know as well. but if you are right, this is the dead code which is not correct. I suppose it is possible, but better check with the developer.

Comment (jcarranzan) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java]: ok I've just discovered how it could work, I've pushed my commit right now.

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (jcarranzan) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:32]: if I remove that line, all the tests will fail. What property do you suggest instead?

Comment (michalvavrik) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:32]: according to the documentation `quarkus.management.enabled` is in order to expose management interface on a different port; just use same port?

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (michalvavrik) [inline monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:18]: please tag this class with your JIRA ticket

Comment (michalvavrik) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:18]: `@Tag(QUARKUS-ABCD)`

File: monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java
Comment (jcarranzan) [reply monitoring/micrometer-prometheus/src/test/java/io/quarkus/ts/micrometer/prometheus/HttpPathMetricsPatternIT.java:32]: Yes, let's remove it and modify it to `String metrics = app.given().get("/q/metrics").then()` as your suggestion, it's simplest and it's valid.

**Review by michalvavrik — APPROVED** [review]
lgtm provided CI is green, good job

**Pull request merged**
